### PR TITLE
Zeus Improvements

### LINKED ISSUE #369: Feature Request - Toggle for Zeus comms when possessing units
Add a control that allows a Zeus player to switch between the following behaviours when Zeus controlling an AI unit:

- All voice and radio comms are through the controlled unit - Useful for role-playing characters talking to non-zeus players
- All voice and radio comms remain through the Zeus player's original body - Useful for talking with other Zeuses and remaining on the radio while controlling enemy AI.

### LINKED ISSUE #376: Improve Zeus support
Inspired by comments in this thread: https://forums.bistudio.com/forums/topic/193813-advanced-combat-radio-environment-2-acre2/?page=20#comment-3185498

- Add the ability to hear players proximity voice when nearby in free camera mode. Currently if Zeus is in free camera mode he can only hear things that are close to his virtual entity or body. He should be able to hear things nearby his floating camera. I believe ACE3 does something to make it work with spectators, might be worth looking into.

- Add the ability to listen in on certain radio channels to know what your players are up to. I think this is currently possible only when Zeus is not a virtual entity.

### PR DESCRIPTION
**Pull request by BrettMayson**
**When merged this pull request will:**
- [x] Close #376 
- [x] Close #369 
- [x] Allow Zeus to join spectator mode
- [x] Enable Zeus to hear nearby players and external radios
- [x] Allow Zeus to talk from the camera location

**TODO Before Merge**
- [x] Medium Scale Multiplayer Testing
- [x] "ACRE Zeus" Settings
- [x] Keybinds

### CODE REVIEW

**Comment by jonpas** [comment]
Awesome!

I'd add keybinds for both switches, unbound by default, for people that Zeus a lot or don't run ACE3. 

Correct me if I am wrong, this retains old functionality and adds proper support for camera etc with those interactions right?

File: addons/sys_core/fnc_remoteStartSpeaking.sqf [hunk 1]
```diff
@@ -102,6 +102,9 @@ private _result = false;
                 _unit setVariable [QGVAR(currentSpeakingRadio), _radioId];
                 private _speakerRadio = [];
                 private _nearRadios = [ACRE_LISTENER_POS, 150] call EFUNC(sys_radio,nearRadios);
+                if (!isNull(findDisplay 312)) then { //Zeus
+                    _nearRadios = _nearRadios + [(getPosASL curatorCamera), 150] call EFUNC(sys_radio,nearRadios);
```
Comment (jokoho48) [inline addons/sys_core/fnc_remoteStartSpeaking.sqf]: should use append is faster.

**Comment by BrettMayson** [comment]
No matter what I do in this PR, I want to make sure everything that woks now works as expected

File: addons/sys_core/fnc_remoteStartSpeaking.sqf (hunk 1 above)
Comment (BrettMayson) [reply addons/sys_core/fnc_remoteStartSpeaking.sqf]: 👌

File: addons/sys_core/fnc_inRange.sqf [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+];
+
+if (_position distance ACRE_LISTENER_POS < _range) exitWith { true };
+
+// check the zeus camera, only with the zeus interface is active
+if (_checkZeus && {!isNull(findDisplay 312)}) then {
+    if (_position distance (getPosASL curatorCamera) < _range) exitWith { true };
+};
+
+false
```
Comment (jokoho48) [inline addons/sys_core/fnc_inRange.sqf]: this function will not work as intended. the main issue is that the exitWith in the zeus check will get overwritten by the false at the end.
and this function can be simplified to
```sqf
if (_checkZeus && {!isNull(findDisplay 312)}) exitWith {
    (_position distance (getPosASL curatorCamera) < _range)
};
(_position distance ACRE_LISTENER_POS < _range)
```

**Comment by BrettMayson** [comment]
>I'd add keybinds for both switches, unbound by default, for people that Zeus a lot or don't run ACE3.

I might change this up as the PR adds in more things, but at the end I will add in keybinds for everything

>adds proper support for camera etc with those interactions right

When spectator is enabled, Zeus will hear voices from both spectator and the camera as expected, and talk in spectator

When spectator is disabled, Zeus will hear voices from their unit and the camera. I am working on a toggle to switch between talking from their unit and talking from the camera.

File: addons/sys_core/fnc_remoteStartSpeaking.sqf [hunk 3]
```diff
@@ -102,6 +102,9 @@ private _result = false;
                 _unit setVariable [QGVAR(currentSpeakingRadio), _radioId];
                 private _speakerRadio = [];
                 private _nearRadios = [ACRE_LISTENER_POS, 150] call EFUNC(sys_radio,nearRadios);
+                if (!isNull(findDisplay 312)) then { //Zeus
```
Comment (jokoho48) [inline addons/sys_core/fnc_remoteStartSpeaking.sqf]: i am not sure but is this faster or just `isNull curatorCamera`?

File: addons/sys_core/fnc_inRange.sqf (hunk 2 above)
Comment (BrettMayson) [reply addons/sys_core/fnc_inRange.sqf]: I knew the exitWith would be overwritten, but I didn't think of that simplification. I usually try to write so that in the usual use case, the least amount of code is executed, hence the Zeus if after the unit exitWith

File: addons/sys_core/fnc_remoteStartSpeaking.sqf (hunk 3 above)
Comment (BrettMayson) [reply addons/sys_core/fnc_remoteStartSpeaking.sqf]: This is what I use in my mod, I am pretty sure I ran tests, but I'll double check

**Comment by BrettMayson** [comment]
Adding radios to zeus is going to require a lot of work in more places, so I'm moving that to a different PR

File: addons/sys_core/XEH_preInit.sqf [hunk 4]
```diff
@@ -99,7 +99,13 @@ acre_sys_io_ioEventFnc = {
 };
 
 ["unit", {
+    acre_current_player = (_this select 0);
```
Comment (TheMagnetar) [inline addons/sys_core/XEH_preInit.sqf]: no ()

File: addons/sys_core/XEH_preInit.sqf [hunk 5]
```diff
@@ -99,7 +99,13 @@ acre_sys_io_ioEventFnc = {
...
     acre_player = (_this select 0);
+    if !(acre_player isEqualTo player) then {
+        if ((_this select 0) getVariable [QEGVAR(sys_zeus,usePlayer), GVAR(defaultZeusVoiceLocation)]) then {
```
Comment (TheMagnetar) [inline addons/sys_core/XEH_preInit.sqf]: `_this select 0` is stored already as `acre_current_player`

File: addons/sys_core/fnc_inRange.sqf [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+params [
+    ["_unit", objNull, [objNull]]
+];
+
+private _position = getPosASL _unit;
+
+private _check = {
+    if (call FUNC(inZeus)) exitWith {
+        (_this distance (getPosASL curatorCamera) < 300)
```
Comment (TheMagnetar) [inline addons/sys_core/fnc_inRange.sqf]: Magic number. It should be a define.

File: addons/sys_core/fnc_inRange.sqf [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    };
+    (_this distance ACRE_LISTENER_POS < 300)
```
Comment (TheMagnetar) [inline addons/sys_core/fnc_inRange.sqf]: Magic number. It should be a define.

File: addons/sys_core/fnc_inRange.sqf [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+};
+
+if (_unit getVariable [QEGVAR(zeus,inZeus), false]) exitWith {
+    ((_unit getVariable [QEGVAR(zeus,zeusPosition), [0,0,0]]) call _check)
```
Comment (TheMagnetar) [inline addons/sys_core/fnc_inRange.sqf]: no wrapping () -> `(_unit getVariable [QEGVAR(zeus,zeusPosition), [0,0,0]]) call _check`

File: addons/sys_core/fnc_inRange.sqf [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+};
+(_position call _check)
```
Comment (TheMagnetar) [inline addons/sys_core/fnc_inRange.sqf]: no ()

File: addons/sys_core/fnc_inZeus.sqf [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+ * Return Value:
+ * In Zeus Interface <BOOL>
+ *
+ * Example:
+ * _inZeus = call acre_sys_core_fnc_inZeus
+ *
+ * Public: No
+ */
+
+!isNull(findDisplay 312)
```
Comment (TheMagnetar) [inline addons/sys_core/fnc_inZeus.sqf]: space between `isNull` and `findDisplay`. Additionally the 312 should be a  `#define`

File: addons/sys_core/fnc_processDirectSpeaker.sqf [hunk 11]
```diff
@@ -39,8 +39,13 @@ if (_bothSpectating || {_isIntercomAttenuate}) then {
     _emitterPos = ACRE_LISTENER_POS;
     _emitterDir = ACRE_LISTENER_DIR;
 } else {
-    _emitterPos = (AGLtoASL (_unit modelToWorldVisual (_unit selectionPosition "head"))); //; eyePos _unit;
-    _emitterDir = eyeDirection _unit;
+    if (_unit getVariable [QEGVAR(zeus,inZeus), false]) then {
+        _emitterPos = _unit getVariable [QEGVAR(zeus,zeusPosition), false];
+        _emitterDir = _unit getVariable [QEGVAR(zeus,zeusDir), false];
+    } else {
+        _emitterPos = (AGLtoASL (_unit modelToWorldVisual (_unit selectionPosition "head"))); //; eyePos _unit;
```
Comment (TheMagnetar) [inline addons/sys_core/fnc_processDirectSpeaker.sqf]: no wrapping () -> `AGLtoASL (_unit modelToWorldVisual (_unit selectionPosition "head"))`

File: addons/sys_core/fnc_remoteStartSpeaking.sqf [hunk 12]
```diff
@@ -102,6 +102,9 @@ private _result = false;
                 _unit setVariable [QGVAR(currentSpeakingRadio), _radioId];
                 private _speakerRadio = [];
                 private _nearRadios = [ACRE_LISTENER_POS, 150] call EFUNC(sys_radio,nearRadios);
+                if (call FUNC(inZeus)) then { //Zeus
+                    _nearRadios append [(getPosASL curatorCamera), 150] call EFUNC(sys_radio,nearRadios);
```
Comment (TheMagnetar) [inline addons/sys_core/fnc_remoteStartSpeaking.sqf]: Magic number. Let's modify  the 150 of the previous line as well.

File: addons/sys_zeus/fnc_handleZeusSpeakPress.sqf [hunk 13]
```diff
@@ -0,0 +1,15 @@
+/*
+ * Author: SynixeBrett
+ * Handles the speak from zeus camera press down
+ *
+ * Public: No
+ */
+#include "script_component.hpp"
+
+if (call EFUNC(sys_core,inZeus) && {!(ACRE_IS_SPECTATOR)}) then {
```
Comment (TheMagnetar) [inline addons/sys_zeus/fnc_handleZeusSpeakPress.sqf]: no need for () -> `!ACRE_IS_SPECTATOR`

**Review by TheMagnetar — CHANGES_REQUESTED** [review]

File: addons/sys_zeus/XEH_postInit.sqf [hunk 14]
```diff
@@ -0,0 +1,17 @@
+#include "script_component.hpp"
+
+private _category = format ["ACRE %1", localize "str_a3_cfghints_curator_curator_displayname"];
+
+[_category, "ZeusTalkFromCamera",  [(LLSTRING(SpeakFromCamera)), (LLSTRING(SpeakFromCamera_description))],
+    { call FUNC(handleZeusSpeakPress) },
+    { call FUNC(handleZeusSpeakPressUp) },
+[40, [false, false, false]]] call cba_fnc_addKeybind; //Default bound to `
+
+// There is no EH for opening the zeus interface
```
Comment (jonpas) [inline addons/sys_zeus/XEH_postInit.sqf]: You can do it via CBA XEH for Displays, like here: https://github.com/acemod/ACE3/blob/e5a15d200f44df5fccc0bc5575d18d80b35538dd/addons/spectator/CfgEventHandlers.hpp#L27-L29
Which is effectively an EH for opening the Zeus interface.

File: addons/sys_zeus/fnc_setUsePlayer.sqf [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+ * set if the player speaks through their player or the remote controlled player
+ *
+ * Arguments:
+ * 0: true to speak through player, false for remote controlled <BOOLEAN>
+ *
+ * Return Value:
+ * Nothing
+ *
+ * Example:
+ * [true] call acre_sys_zeus_fnc_setUsePlayer;
```
Comment (jonpas) [inline addons/sys_zeus/fnc_setUsePlayer.sqf]: No final semi-colon.

File: addons/sys_zeus/fnc_setUsePlayer.sqf [hunk 16]
```diff
@@ -0,0 +1,30 @@
+/*
+ * Author: SynixeBrett
+ * set if the player speaks through their player or the remote controlled player
```
Comment (jonpas) [inline addons/sys_zeus/fnc_setUsePlayer.sqf]: `Set`, missing final dot.

File: addons/sys_zeus/fnc_setUsePlayer.sqf [hunk 17]
```diff
@@ -0,0 +1,30 @@
...
+ *
+ * Arguments:
+ * 0: true to speak through player, false for remote controlled <BOOLEAN>
```
Comment (jonpas) [inline addons/sys_zeus/fnc_setUsePlayer.sqf]: ` * 0: Speak through player (true) or remote controlled unit (false) <BOOL>`

File: addons/sys_zeus/fnc_setUsePlayer.sqf [hunk 18]
```diff
@@ -0,0 +1,30 @@
...
+ *
+ * Return Value:
+ * Nothing
```
Comment (jonpas) [inline addons/sys_zeus/fnc_setUsePlayer.sqf]: ` * None`

File: addons/sys_zeus/fnc_handleZeusSpeakPressUp.sqf [hunk 19]
```diff
@@ -0,0 +1,10 @@
+/*
+ * Author: SynixeBrett
+ * Handles the speak from zeus camera press up
```
Comment (jonpas) [inline addons/sys_zeus/fnc_handleZeusSpeakPressUp.sqf]: Missing final dot and the rest of the header content.

File: addons/sys_zeus/fnc_handleZeusSpeakPress.sqf [hunk 20]
```diff
@@ -0,0 +1,15 @@
+/*
+ * Author: SynixeBrett
+ * Handles the speak from zeus camera press down
```
Comment (jonpas) [inline addons/sys_zeus/fnc_handleZeusSpeakPress.sqf]: Missing final dot and the rest of the header content.

File: addons/sys_zeus/fnc_handleZeusSpeakPress.sqf [hunk 21]
```diff
@@ -0,0 +1,15 @@
...
+ *
+ * Public: No
+ */
+#include "script_component.hpp"
+
+if (call EFUNC(sys_core,inZeus) && {!ACRE_IS_SPECTATOR}) then {
+    GVAR(speakFromZeusHandle) = [{
+        player setVariable [QGVAR(inZeus), true, true];
```
Comment (jonpas) [inline addons/sys_zeus/fnc_handleZeusSpeakPress.sqf]: Do those variables really have to be synced over network?

File: addons/sys_zeus/ACE_Actions.hpp [hunk 22]
```diff
@@ -0,0 +1,17 @@
+class ACE_ZeusActions {
```
Comment (jonpas) [inline addons/sys_zeus/ACE_Actions.hpp]: Might be cleaner if we add all actions via SQF under "if ACE running...", would also lower size and amount of files.

Look at how racks and intercom do it.

**Review by jonpas — CHANGES_REQUESTED** [review]
Finally proper mid-review.

**Review by TheMagnetar — APPROVED** [review]

File: addons/sys_core/XEH_preInit.sqf [hunk 23]
```diff
@@ -99,7 +99,13 @@ acre_sys_io_ioEventFnc = {
 };
 
 ["unit", {
-    acre_player = (_this select 0);
+    acre_current_player = _this select 0;
```
[the remark this answers is not in the document]
Comment (BrettMayson) [reply addons/sys_core/XEH_preInit.sqf:108]: I'm not sure what else to use then, is there something you would prefer? This variable will always point to the currently controlled unit, where as acre_player will now point to the unit being used for ACRE, which may not be the current unit.

File: addons/sys_core/fnc_inRange.sqf [hunk 24]
```diff
@@ -0,0 +1,36 @@
...
+ * In range <BOOL>
+ *
+ * Example:
+ * _unit call acre_sys_core_fnc_inRange
+ *
+ * Public: No
+ */
+#include "script_component.hpp"
+
+#define MAX_DIRECT_RANGE 300
```
[the remark this answers is not in the document]
Comment (BrettMayson) [reply addons/sys_core/fnc_inRange.sqf]: Can do, I looked around and noticed [this](https://github.com/IDI-Systems/acre2/blob/master/addons/sys_core/fnc_enableRevealAI.sqf#L18) so I put in here. Out of scope for this PR, but should all the magic number defines be moved to script_component?

File: addons/sys_zeus/fnc_handleZeusSpeakPress.sqf (hunk 21 above)
Comment (jonpas) [reply addons/sys_zeus/fnc_handleZeusSpeakPress.sqf]: Ah I see. That's quite a lot of network traffic. Could we combine those 3 datas into one for starters?

File: addons/sys_core/XEH_preInit.sqf (hunk 23 above)
[the remark this answers is not in the document]
Comment (jonpas) [reply addons/sys_core/XEH_preInit.sqf:108]: Hmm, I am not sure how safe it is to change acre_player due to API functions and so on. 

So acre_current_player is what acre_player used to be? I'd rather keep that and use new variable for it (name maybe "acre_source_unit").

That way we make sure all API functions keep working, just need to change what is connected with speaking.

File: addons/sys_zeus/fnc_handleZeusSpeakPress.sqf (hunk 21 above)
Comment (BrettMayson) [reply addons/sys_zeus/fnc_handleZeusSpeakPress.sqf]: I noticed `inZeus` can be moved outside perFrameHandler, and I've combined the other two in one

File: addons/sys_core/XEH_preInit.sqf (hunk 23 above)
[the remark this answers is not in the document]
Comment (jonpas) [reply addons/sys_core/XEH_preInit.sqf:108]: Ok, I like this! Let's leave it as is!

[the remark this answers is not in the document]
Comment (TheMagnetar) [reply addons/sys_core/XEH_preInit.sqf:108]: Wait, we won't be able to use "AI" radios? There are plenty of "roleplay" situations were as AI you may want to get into a vehicle, use the new rack, transmit coordinates for extraction and so on... I don't use this personally but there may be communities that may use this..

[the remark this answers is not in the document]
Comment (BrettMayson) [reply addons/sys_core/XEH_preInit.sqf:108]: You are able to use AI radios, that is how it works now, so I have maintained that as default.
You can use a setting to change the default behavior to not use AI. https://github.com/synixebrett/acre2/blob/zeus/addons/sys_core/initSettings.sqf#L131

While in remote control you are able to use either a keybind or ACE Self Interaction to switch between the AI and your player unit.

[the remark this answers is not in the document]
Comment (TheMagnetar) [reply addons/sys_core/XEH_preInit.sqf:108]: Perfect then.

File: addons/sys_zeus/fnc_handleZeusSpeakPressUp.sqf [hunk 25]
```diff
@@ -15,11 +15,12 @@
...
-private _wasSpectator = player getVariable [QGVAR(wasSpectator), ACRE_IS_SPECTATOR];
-player setVariable [QGVAR(wasSpectator), ACRE_IS_SPECTATOR];
-[_wasSpectator] call EFUNC(api,setSpectator);
+    // Restore previous out of Zeus state
+    [player getVariable [QGVAR(wasSpectator), ACRE_IS_SPECTATOR]] call EFUNC(api,setSpectator);
 
-[GVAR(speakFromZeusHandle)] call CBA_fnc_removePerFrameHandler;
+    // Stop updating Zeus position
+    [GVAR(speakFromZeusHandle)] call CBA_fnc_removePerFrameHandler;
+}] call CBA_fnc_waitUntilAndExecute;
```
[the remark this answers is not in the document]
Comment (BrettMayson) [reply addons/sys_zeus/fnc_handleZeusSpeakPressUp.sqf:30]: If the button is pushed quickly, it doesn't properly stop talking through the camera. I am still working to fix this.

File: addons/sys_core/fnc_inRange.sqf (hunk 24 above)
[the remark this answers is not in the document]
Comment (jonpas) [reply addons/sys_core/fnc_inRange.sqf]: Yeah, except if they are really related to that only and shouldn't be changed ever.

File: addons/sys_zeus/XEH_preInit.sqf [hunk 26]
```diff
@@ -0,0 +1,40 @@
...
+PREP_RECOMPILE_START;
+#include "XEH_PREP.hpp"
+PREP_RECOMPILE_END;
+
+GVAR(keyDownWait) = false;
+
+private _zeusCategory = format ["ACRE %1", localize "str_a3_cfghints_curator_curator_displayname"];
+
+// Default remote controlled voice source
+[
```
Comment (jonpas) [inline addons/sys_zeus/XEH_preInit.sqf]: Put into `initSettings.sqf` and include.

**Review by jonpas — CHANGES_REQUESTED** [review]

File: addons/sys_zeus/fnc_handleZeusSpeakPressUp.sqf [hunk 27]
```diff
@@ -15,12 +15,13 @@
  * Public: No
  */
 
-[{!(GVAR(keyDownWait))}, {
-    player setVariable [QGVAR(inZeus), false, true];
+if (!canSuspend) exitWith {0 spawn FUNC(handleZeusSpeakPressUp)};
 
-    // Restore previous out of Zeus state
-    [player getVariable [QGVAR(wasSpectator), ACRE_IS_SPECTATOR]] call EFUNC(api,setSpectator);
+sleep 0.5;
```
Comment (jonpas) [inline addons/sys_zeus/fnc_handleZeusSpeakPressUp.sqf]: No.

Why is this necessary? Does it require just a frame delay? Does it require a specific delay? What is that dependent on?

Comment (BrettMayson) [reply addons/sys_zeus/fnc_handleZeusSpeakPressUp.sqf]: I don't like this either, but every other solution I tried didn't work.

Basically, if the key is pressed really quickly, it would start transmitting but never stop. I am very much open to a better solution, I just couldn't get any others to work consistently.

File: addons/sys_gui/fnc_onInventoryRadioDoubleClick.sqf [hunk 28]
```diff
@@ -19,4 +19,7 @@ TRACE_1("enter", _this);
 
 params ["_object","_container","_radioId"];
 
+// When a radio is opened set acre_player to the current player
+acre_player = acre_current_player;
```
Comment (jonpas) [inline addons/sys_gui/fnc_onInventoryRadioDoubleClick.sqf:23]: This is not the only way of opening radios, there are also keybind and ACE Self-Interaction. What is the goal here?

File: addons/sys_zeus/XEH_postInit.sqf [hunk 29]
```diff
@@ -0,0 +1,26 @@
+#include "script_component.hpp"
+
+private _category = format ["ACRE2 %1", localize "str_a3_cfghints_curator_curator_displayname"];
+
+[_category, "ZeusTalkFromCamera",  [(LLSTRING(SpeakFromCamera)), (LLSTRING(SpeakFromCamera_description))],
+    { call FUNC(handleZeusSpeakPress) },
+    { call FUNC(handleZeusSpeakPressUp) },
+[41, [false, false, false]]] call CBA_fnc_addKeybind; //Default bound to `
```
Comment (jonpas) [inline addons/sys_zeus/XEH_postInit.sqf]: Can probably come up with a better default. Zeus doesn't use as many keys after-all.

File: addons/sys_zeus/XEH_postInit.sqf [hunk 30]
```diff
@@ -0,0 +1,26 @@
...
+
+if (hasInterface && {isClass (configFile >> "CfgPatches" >> "ace_interact_menu")}) then {
+    private _acreNode = ["ACRE_ZeusEars", "ACRE", "", {}, { true }] call ace_interact_menu_fnc_createAction;
```
Comment (jonpas) [inline addons/sys_zeus/XEH_postInit.sqf]: "ACRE" could be "Ears" or something like that. Depends if we want to add more stuff I guess.

File: addons/sys_zeus/fnc_handleZeusSpeakPressUp.sqf (hunk 25 above)
[the remark this answers is not in the document]
Comment (jonpas) [reply addons/sys_zeus/fnc_handleZeusSpeakPressUp.sqf:30]: Let's make a separate issue for that to figure it out one day.

**Review by jonpas — CHANGES_REQUESTED** [review]
Looking great, just some smaller notes and a question. Otherwise this is good to merge.

> Only minor bug is that listening to players through the camera doesn't work on the carrier. The players can hear Zeus if they talk through the camera, but Zeus can't hear them.

Also put this into separate issue to keep track as known issue.

File: addons/sys_zeus/XEH_postInit.sqf (hunk 29 above)
Comment (BrettMayson) [reply addons/sys_zeus/XEH_postInit.sqf]: My group uses this almost every single OP now. In my opinion, it's a pretty good key.

File: addons/sys_gui/fnc_onInventoryRadioDoubleClick.sqf (hunk 28 above)
Comment (BrettMayson) [reply addons/sys_gui/fnc_onInventoryRadioDoubleClick.sqf:23]: When remote controlling a unit the interaction menu will use their unit's radios instead of the remote-controlled unit's radios. This will switch to the remote-controlled unit's radios if they open a radio via the inventory.

Comment (BrettMayson) [reply addons/sys_gui/fnc_onInventoryRadioDoubleClick.sqf:23]: This will be improved when I do the PR for Zeus radios since right now it is unclear what radios Zeus will be using, especially if they are a virtual entity and not a regular unit.

File: addons/sys_zeus/XEH_postInit.sqf (hunk 29 above)
Comment (jonpas) [reply addons/sys_zeus/XEH_postInit.sqf]: On my keyboard that's AltGr+7, rather hard to access.

File: addons/sys_gui/fnc_onInventoryRadioDoubleClick.sqf (hunk 28 above)
Comment (jonpas) [reply addons/sys_gui/fnc_onInventoryRadioDoubleClick.sqf:23]: Ah I see, and that's not a problem when opening with keybind etc. Sounds good.

**Review by jonpas — APPROVED** [review]
Very happy with this iteration!
